os/binfmt/libelf/libelf_bind.c

```diff
@@ -194,32 +195,40 @@ static int elf_relocate(FAR struct elf_loadinfo_s *loadinfo, int relidx, FAR con
 
 	for (i = 0; i < relsec->sh_size / sizeof(Elf32_Rel); i++) {
 		psym = &sym;
+		prel = &rel;
+
 
 		/* Read the relocation entry into memory */
 
-		ret = elf_readrel(loadinfo, relsec, i, &rel);
-		if (ret < 0) {
-			berr("Section %d reloc %d: Failed to read relocation entry: %d\n", relidx, i, ret);
-			goto ret_err;
+		if (loadinfo->reltab) {
+			prel = loadinfo->reltab + sizeof(Elf32_Rel) * i;
+		} else {
+			ret = elf_readrel(loadinfo, relsec, i, &rel);
+			if (ret < 0) {
+				berr("Section %d reloc %d: Failed to read relocation entry: %d\n", relidx, i, ret);
+				goto ret_err;
+			}
 		}
 
 		/* Get the symbol table index for the relocation.  This is contained
 		 * in a bit-field within the r_info element.
 		 */
 
-		symidx = ELF32_R_SYM(rel.r_info);
+		symidx = ELF32_R_SYM(prel->r_info);
 
 		/* Read the symbol table entry into memory */
-
-		ret = elf_readsym(loadinfo, symidx, &sym);
-		if (ret < 0) {
-			berr("Section %d reloc %d: Failed to read symbol[%d]: %d\n", relidx, i, symidx, ret);
-			goto ret_err;
+		if (loadinfo->symtab) {
+			psym = loadinfo->symtab + sizeof(Elf32_Sym) * symidx;
```

Comment:
Pls check index out of bounds condition for symtab.

Reply:
Done. Please check.